src/Workspaces/CSharp/Portable/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -195,6 +196,13 @@ private static CSharpSyntaxContext CreateContextWorker(Workspace workspace, Sema
                                             targetToken.Parent.IsKind(SyntaxKind.DestructorDeclaration) &&
                                             targetToken.Parent.Parent.IsKind(SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration);
 
+            // Typing a dot after a numeric expression (numericExpression.) 
+            // - maybe a start of MemberAccessExpression like numericExpression.Member.
+            // - or it maybe a start of a range expression like numericExpression..anotherNumericExpression (starting C# 8.0) 
+            // Therefore, in the scenario, we want the completion to be __soft selected__ until user types the next character after the dot.
+            // If the second dot was typed, we just insert two dots.
+            var isSoftSelectionContext = IsRightSideOfNumericType(leftToken, semanticModel, cancellationToken);
```

Comment:
this hsould just be `var isRightSideOfNumericType` and that should be the field/param name as well.